pkg/operator/cluster.go

```diff
@@ -77,23 +93,50 @@ func (t *clusterTPR) findAllClusters() error {
 	logger.Infof("found %d clusters", len(clusterList.Items))
 	for i := range clusterList.Items {
 		c := clusterList.Items[i]
+		t.startCluster(&c)
+	}
 
-		ns := c.Spec.Namespace
-		existingCluster := cluster.New(c.Spec, t.context.factory, t.context.clientset)
-		t.tracker.add(ns, c.Metadata.ResourceVersion)
-		t.clusters[ns] = existingCluster
+	t.watchVersion = clusterList.Metadata.ResourceVersion
+	return nil
+}
 
-		logger.Infof("resuming cluster %s in namespace %s", c.Metadata.Name, ns)
-		t.startCluster(existingCluster)
+func (t *clusterManager) startTrack(c *cluster.Cluster) error {
+	t.Lock()
+	defer t.Unlock()
+
+	existing, ok := t.clusters[c.Namespace]
+	if ok {
+		if c.Name != existing.Name {
+			return fmt.Errorf("cluster %s already running in namespace %s. multiple clusters per namespace not supported", existing.Name, existing.Namespace)
+		}
+	} else {
+		// only start the cluster if we're not already tracking it from a previous iteration
+		t.clusters[c.Namespace] = c
 	}
 
-	t.watchVersion = clusterList.Metadata.ResourceVersion
+	// refresh the version of the cluster we're tracking
+	t.tracker.add(c.Namespace, c.ResourceVersion)
+
 	return nil
 }
 
-func (t *clusterTPR) startCluster(c *cluster.Cluster) {
+func (t *clusterManager) stopTrack(c *cluster.Cluster) {
+	t.Lock()
+	defer t.Unlock()
+
+	t.tracker.remove(c.Namespace)
+	delete(t.clusters, c.Namespace)
+}
+
+func (t *clusterManager) startCluster(c *cluster.Cluster) {
+	c.Init(t.context.factory, t.context.clientset)
+	if err := t.startTrack(c); err != nil {
+		logger.Errorf("failed to start cluster %s in namespace %s. %+v", c.Name, c.Namespace, err)
+		return
+	}
+
 	if t.devicesInUse && c.Spec.Storage.AnyUseAllDevices() {
-		logger.Warningf("devices in more than one namespace not supported. ignoring devices in namespace %s", c.Spec.Namespace)
+		logger.Warningf("devices in more than one namespace not supported. ignoring devices in namespace %s", c.Namespace)
```

Comment:
i think this was my warning message and it doesn't quite make sense.  if you are making other updates for this PR, do you want to also change this to something like:
> using all devices in more than one namespace...